common/lib/api-client/middleware/request.js

```diff
@@ -0,0 +1,39 @@
+const { get } = require('lodash')
```

Comment:
So complexity wise I have had a little looks at it. There may be bits here that help:

```
const { get } = require('lodash')

const redisStore = require('../../../../config/redis-store')()
const models = require('../models')

function requestMiddleware(expiry = 60) {
  async function req(payload) {
    const { req, jsonApi } = payload
    const pathname = new URL(req.url).pathname
    const searchString = Object.keys(req.params).length
      ? `?${new URLSearchParams(req.params).toString()}`
      : ''
    const key = `cache:${req.method}.${pathname}${searchString}`
    const hasCacheOption = Boolean(
      get(models, `${req.model}.options.cache`, false)
    )

    if (!hasCacheOption) {
      return jsonApi.axios(req)
    }

    return redisStore.client.getAsync(key).then(response => {
      if (response) {
        return { data: JSON.parse(response) }
      }

      return jsonApi.axios(req).then(async response => {
        await redisStore.client.setexAsync(
          key,
          expiry,
          JSON.stringify(response.data)
        )
        return response
      })
    })
  }

  return {
    name: 'axios-request',
    req,
  }
}

module.exports = requestMiddleware
```